File: rs2-rent-sistem/rs2-rent-sistem.Services/Database/Damage.cs
using System;
using System.Collections.Generic;

namespace rs2_rent_sistem.Services.Database;

public partial class Damage
{
    public int ID { get; set; }

    public string? Comment { get; set; }

    public DateTime DateAdded { get; set; }

    public int OrderItemID { get; set; }

    public virtual OrderItem OrderItem { get; set; }
}

File: rs2-rent-sistem/rs2-ui/rs2_rent_sistem/lib/models/damage/damage.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:rs2_rent_sistem/models/item_in_order/order_item.dart';

part 'damage.freezed.dart';
part 'damage.g.dart';

@Freezed()
class Damage with _$Damage {
  const factory Damage({
    required int id,
    required ItemInOrder orderItem,
    @Default('') String comment,
    DateTime? dateAdded,
  }) = _Damage;

  factory Damage.fromJson(Map<String, dynamic> json) => _$DamageFromJson(json);
}